Sources/Networking/Caching/CustomerInfoCallback.swift

```diff
@@ -28,3 +28,41 @@ struct CustomerInfoCallback: CacheKeyProviding {
     }
 
 }
+
+extension CustomerInfoCallback {
+
+    func withNewCacheKey(_ newKey: String) -> Self {
+        var copy = self
+        copy.cacheKey = newKey
+
+        return copy
+    }
+
+}
+
+// MARK: - CallbackCache helpers
+
+extension CallbackCache where T == CustomerInfoCallback {
+
+    func addOrAppendToPostReceiptDataOperation(callback: CustomerInfoCallback) -> CallbackCacheStatus {
+        if let existing = self.callbacks(ofType: PostReceiptDataOperation.self).last {
+            return self.add(callback: callback.withNewCacheKey(existing.cacheKey))
+        } else {
+            return self.add(callback: callback)
+        }
+    }
+
+    private func callbacks(ofType type: NetworkOperation.Type) -> [T] {
+        return self
+            .cachedCallbacksByKey
+            .value
+            .lazy
+            .flatMap(\.value)
+            .filter { $0.source == type }
+    }
+
+    private func callbackCount(ofType type: NetworkOperation.Type) -> Int {
```

Comment:
This seems unused?

Reply:
Oh yeah that was for the old implementation. Thanks.